pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -299,12 +287,28 @@ func buildHTTPVirtualServices(obj config.Config, gateways map[parentKey]map[k8s.
 
 		route, err := buildHTTPDestination(r.BackendRefs, ns, domain, zero)
 		if err != nil {
-			reportError(err)
-			return
+			return err
 		}
 		vs.Route = route
 
 		httproutes = append(httproutes, vs)
+		return nil
+	}
+
+	for _, r := range route.Rules {
+		if len(r.Matches) > 1 {
+			continue
```

Comment:
nvm, I get it now. Maybe it would be simpler if we remove the splitRules list above and then do it all in one place:

```
	for _, r := range route.Rules {
		if len(r.Matches) > 1 {
			// split the rule to make sure each rule has up to one match
			matches := r.Matches
			for _, m := range matches {
				r.Matches = []k8s.HTTPRouteMatch{m}
				if err := convertHTTPRoute(r); err != nil {
					reportError(err)
					return
				}
			}
		} else {
			if err := convertHTTPRoute(r); err != nil {
				reportError(err)
				return
			}
		}
	}
```

Right now the code is spread out and seems like it could easily break in refactor, etc, and a bit hard to follow. This is also (trivially) more efficient to not store in a list